network-mux/src/Network/Mux.hs

```diff
@@ -232,10 +237,20 @@ muxChannel tracer pmss mid md cnt = do
                 (printf "Attempting to send a message of size %d on %s %s" (BL.length encoding)
                         (show mid) (show $ md))
                 callStack
-        atomically $ modifyTVar cnt (+ 1)
-        atomically $ putTMVar w encoding
+
         traceWith tracer $ MuxTraceChannelSendStart mid encoding
-        atomically $ writeTBQueue (tsrQueue pmss) (TLSRDemand mid md want)
+
+        atomically $ do
+            buf <- readTVar w
+            if BL.length buf < perMiniProtocolBufferSize
```

Comment:
`perMiniProtocolBufferSize` seems to be to do rate limiting of writes for a mini-protocol.  In the future we might experiment with different values per mini-protocol: chian-sync just sends headers but block-fetch sends whole blocks. I think ideally we would have it proportional with respect to header / block size. What do you think @karknu?

Reply:
Previously there was a limit of one message per miniprotocol. When I moved from a TMVar to a TVar I wanted to still have some kind of limit to constrain the memory consumption. A miniprotocol can still have at least one message in flight (we can still cope with a 1Mbyte block message).
I don't see a point in making this any more complicated than it is. The buffer only has to be larger than a maximum mux segment so that there is a full message ready to go once `mux` thread is ready to service the miniprotocol again.